satpy/composites/__init__.py

```diff
@@ -275,7 +270,7 @@ def infer_mode(cls, data_arr):
             return data_arr.attrs['mode']
         if 'bands' not in data_arr.dims:
             return cls.modes[1]
-        if 'bands' in data_arr.coords and isinstance(data_arr.coords['bands'][0], str):
+        if 'bands' in data_arr.coords and isinstance(data_arr.coords['bands'][0].item(), str):
```

Comment:
So was this ever working?

Reply:
Maybe it was with previous versions xarray? If there was such a time that coordinates could be strings?